Tasks/AzureResourceGroupDeploymentV2/operations/ResourceGroup.ts

```diff
@@ -181,7 +181,20 @@ export class ResourceGroup {
                         policyLink = this.getPolicyHelpLink(error.details[i]);
                         errorMessage = this.getPolicyErrorMessage(error.details[i]);
                     } else {
-                        errorMessage = util.format("%s: %s %s", error.details[i].code, error.details[i].message, error.details[i].details);
+                        errorMessage = util.format("%s: %s", error.details[i].code, error.details[i].message);
+                        if(error.details[i].details != null && error.details[i].details != undefined) {
+                            if(typeof error.details[i].details == 'object') {
+                                if(Array.isArray(error.details[i].details)) {
+                                    for(var detail in error.details[i].details) {
+                                        errorMessage += "\n" + JSON.stringify(detail);
+                                    }
+                                } else {
+                                    errorMessage += "\n" + JSON.stringify(error.details[i].details);
+                                }
+                            } else {
+                                errorMessage += " " + String(error.details[i].details);
```

Comment:
Better do error.details[i].details.toString()

Reply:
This seemed safer, there are some objects like undefined/null that donot have .toString() method in javascript. We're already handling null and undefined, but I was worried if there could be a case we're not aware.